### ISSUE TITLE
Slow/infinte loop symbol matching on multiline functions

### ISSUE THREAD

**Issue by amauragis** [Gimly/vscode-matlab#25]
Consider this valid, potentially questionable function declaration:
```
function [example, foo, bar, baz, supercalifragilisticexpialidocious] = ...
         thing(arg1, arg2)
  % body
end
```

A vscode process will eat a core trying to resolve the symbols.

I tested your regex: `/function *(?:(?:(?:\[(?: *\w+ *,? *)+\])|(?:\w+)) *= *)?(?:\w.?)+ *\(.*?\)/` [on the internet](https://regex101.com/r/zpHdpD/1) and it times out.

To reproduce, copy said snippet into a file that the extension tries to parse.

At minimum, we should probably add a timeout to the regex so we don't eat a core, even if we don't support that weird function syntax

**Comment by Gimly** [comment]
I did a few test. If I test only the first par of the regex:

```regex
function *(?:(?:(?:\[(?: *\w+ *,? *)+\])|(?:\w+)) *= *)
```

Then it's working fine, no timeout, but as soon as I add something after (even just a static value) then it breaks and I get a timeout.

I'm not quite sure what's going on, I have to analyse it further.

**Comment by Gimly** [comment]
I've simplified the regex, and now it shouldn't hang anymore. It will be also way faster to execute.

It could have a few false positive, if the array is incorrectly written, but I don't think this is a very big deal, after all it's still a function, and the linter will complain about the incorrectly written code.

**Comment by Gimly** [comment]
Fixed in version 0.8.0
